src/Layer.js

```diff
@@ -513,15 +499,45 @@ export default class Layer {
         }
     }
 
-    _getViewport () {
-        if (this._integrator) {
-            return this._integrator.renderer.getBounds();
+    _setZoomCenter (matrix) {
+        if (matrix) {
+            // Compute the zoom and center from the matrix.
+            // This is a solution to avoid subscribing to map events and
+            // make a better and efficient use of the Custom Layers interface.
+            // TODO: the best solution is to use the matrix at the shader
+            // level and remove the aspect and scale logic from the renderer
+            this.renderer.setCenter(
+                -(1 + 2 * matrix[12] / matrix[0]),
+                +(1 + 2 * matrix[13] / matrix[5])
+            );
+            this.renderer.setZoom(
+                -(2 * matrix[15] / matrix[5])
+            );
+        } else {
+            this._setRendererZoom();
+            this._setRendererCenter();
         }
     }
+
+    _setRendererZoom () {
+        const b = this.map.getBounds();
+        const nw = b.getNorthWest();
+        const sw = b.getSouthWest();
+        const z = (util.projectToWebMercator(nw).y - util.projectToWebMercator(sw).y) / util.WM_2R;
+        this.renderer.setZoom(z);
+    }
+
+    _setRendererCenter () {
+        const c = this.map.getCenter();
+        this.renderer.setCenter(c.lng / 180.0, util.projectToWebMercator(c).y / util.WM_R);
```

Comment:
Use constant instead of a number (`180.0`)

Reply:
This is legacy code. It can be done with projectToWebMercator(c).x, etc but this is shorter. However, I'll move this to an utils file